modules/solr/src/main/java/org/testcontainers/containers/SolrContainer.java

```diff
@@ -0,0 +1,130 @@
+package org.testcontainers.containers;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.naming.ConfigurationException;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+
+import lombok.SneakyThrows;
+
+import org.apache.commons.lang.StringUtils;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+/**
+ * SolrContainer allows a solr container to be launched and controlled.
+ *
+ * @author Simon Schneider
+ */
+public class SolrContainer<SELF extends SolrContainer<SELF>> extends GenericContainer<SELF> {
```

Comment:
Could be simplified?

```suggestion
public class SolrContainer extends GenericContainer<SolrContainer> {
```

Sorry that the self-typing is a bit fiddly...